components/alert_banner.tsx

```diff
@@ -30,57 +43,89 @@ const AlertBanner = ({
-                        'icon-check': mode === 'success',
-                        'icon-alert-circle-outline': mode === 'info',
-                    })}
-                />
-            </div>
-        )}
-        <div className='AlertBanner__body'>
-            {title && <div className='AlertBanner__title'>{title}</div>}
-            {message && (
-                <div
-                    className={classNames({
-                        AlertBanner__message: Boolean(title),
-                    })}
-                >
-                    {message}
-                </div>
+}: AlertBannerProps) => {
+    const {formatMessage} = useIntl();
+    const bannerIcon = useCallback(() => {
+        if (mode === 'danger' || mode === 'warning') {
+            return (
+                <AlertOutlineIcon
+                    size={24}
+                    color={'currentColor'}
+                />);
+        } else if (mode === 'success') {
+            return (
+                <CheckIcon
+                    size={24}
+                    color={'currentColor'}
+                />);
+        }
+        return (
+            <InformationOutlineIcon
+                size={24}
+                color={'currentColor'}
+            />);
+    }, [mode]);
+
+    return (
+        <div
+            className={classNames(
+                'AlertBanner',
+                mode,
+                className,
+                `AlertBanner--${variant}`,
             )}
-            {children}
-            {(actionButtonLeft || actionButtonRight) && (
-                <div className='AlertBanner__actionButtons'>
-                    {actionButtonLeft}
-                    {actionButtonRight}
+        >
+            {!hideIcon && (
+                <div className='AlertBanner__icon'>
+                    {bannerIcon()}
                 </div>
             )}
+            <div className='AlertBanner__body'>
+                {title && <div className='AlertBanner__title'>{title}</div>}
+                {message && (
+                    <div
+                        className={classNames({
+                            AlertBanner__message: Boolean(title),
+                        })}
+                    >
+                        {message}
+                    </div>
+                )}
+                {children}
+                {(actionButtonLeft || actionButtonRight) && (
+                    <div className='AlertBanner__actionButtons'>
+                        {actionButtonLeft}
+                        {actionButtonRight}
+                    </div>
+                )}
+            </div>
+            {onDismiss && (
+                <OverlayTrigger
+                    trigger={['hover', 'focus']}
+                    delayShow={Constants.OVERLAY_TIME_DELAY}
+                    placement='left'
+                    overlay={closeBtnTooltip || (
+                        <Tooltip id='alert_banner_close_btn_tooltip'>
```

Comment:
Dom `id`s are supposed to be unique within the document. Under sort of rare conditions, we can have more than one alert banner simultaneously. Can we make this id a computed value that is unique and stable per instance, maybe something like

```tsx
const [tooltipId] = useState(`alert_banner_close_btn_tooltip_${Math.random()}`);
...
<Tooltip id={tooltipId} ...
```

We'd only want to compute the value once and never update it, so we wouldn't need `setTooltipId`.

It doesn't look like we can use any of the props to reliably construct a stable id because they are optional and are `ReactNode`s and not necessarily strings.

Reply:
While i do agree on the part that id has to be unique in the document. For this component `Tooltip` however i dont see why they need to have an id in first place? I dont recall having e2e test that depend on it or am i missing something?